runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -1017,21 +1293,230 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
+		overhead = (slope * (double)pgcPauseTimeMs) + ((2.0 * xmaxpct) - xminpct);
+		overhead = OMR_MAX(overhead, 0.0);	
+		/* Expanding simply because pgc time is very small is not a good idea, so return xmaxpct, so that if the pgc cpu overhead wants to expand, only then eden expands */
+		overhead = OMR_MIN(overhead, xmaxpct);
+	}
+
+	return overhead;
+}
+
+double 
+MM_SchedulingDelegate::calculateHybridEdenOverhead(MM_EnvironmentVLHGC *env, uintptr_t pgcPauseTimeMs, double overhead)
+{
+	/* 
+	 * When trying to size eden, there is a delicate balance between pgc overhead (here, overhead is cpu %, or % of time that pgc is active 
+	 * versus inactive -> ex: pgc = 100ms, over 1000ms, overhead = 10%). In certain applications, with certain allocation patterns/liveness, 
+	 * pgc average time may be negatively impacted by growing eden unbounded. 
+	 * This function blends the pgc average time (whether it be the actual pgc historic time, or a "predicted" pgc pause time, is left up to the caller) with overhead (% of time gc is active relative to mutator).
+	 * This strikes a much better balance between pgc pause times, and gc cpu overhead, than if just cpu overhead was used.
+	 * 
+	 * By mapping a pgc time to a corresponding overhead (% of time gc is active relative to mutator), eden sizing logic can make a decision as to whether 
+	 * it wants to contract/expand, based on how much it will change the overhead and pgc times.
+	 */
+	double actualPGCOverheadWeight = 0.5;
+	Assert_MM_true((overhead >= 0.0) && (overhead <= 1.0));
+	double pgcTimeOverhead = mapPgcPauseOverheadToPgcCPUOverhead(env, pgcPauseTimeMs);
+	double hybridEdenOverheadHundredBased = MM_Math::weightedAverage(overhead * 100, pgcTimeOverhead, actualPGCOverheadWeight);
+	return hybridEdenOverheadHundredBased / 100;	
+}
+
+void 
+MM_SchedulingDelegate::adjustIdealEdenRegionCount(MM_EnvironmentVLHGC *env) 
+{
+
+	intptr_t edenChange = _edenSizeFactor;
+	/* Be clear that we have already consumed _edenSizeFactor */
+	_edenSizeFactor = 0;
+
+	if (!_extensions->statupPhaseFinished) {
+		/* If currently in startup phase, eden size is being driven by a different set of heuristics - see MM_SchedulingDelegate::heapReconfigured() */
+		return;
+	}
+
+	uintptr_t maxEdenCount = (uintptr_t)(_numberOfHeapRegions * _maxEdenPercent);
+	uintptr_t minEdenCount = (uintptr_t)(_numberOfHeapRegions * _minEdenPercent);
+
+	/* If there are any user specific eden sizing options, these take precendence over _maxEdenPercent and  _minEdenPercent */
+	if (_extensions->userSpecifiedParameters._Xmn._wasSpecified || _extensions->userSpecifiedParameters._Xmns._wasSpecified) {
+		minEdenCount = _extensions->tarokIdealEdenMinimumBytes / _regionManager->getRegionSize();
+	}
+	if (_extensions->userSpecifiedParameters._Xmn._wasSpecified || _extensions->userSpecifiedParameters._Xmnx._wasSpecified) {
+		maxEdenCount = _extensions->tarokIdealEdenMaximumBytes / _regionManager->getRegionSize();
+	}
```

Comment:
I know this is cheap to recompute on each GC cycle, but I believe these 2 Xmn/Xmnx checks could be done once during SchedulingDelegate initialization and where would update _maxEdenPercent/_minEdenPercent